feat: Added functionality for applying and reseting settings

SmolinIvan opened the pull request.

SmolinIvan committed:
feat: Added functionality for applying and reseting settings

SmolinIvan committed:
Test fixes

mikebars1995 commented on src/index.tsx:
```diff
@@ -13,19 +21,60 @@ const domNode = document.getElementById('root') as HTMLDivElement;
 const root = createRoot(domNode);
 
 const App = () => {
```
**Можно лучше**

* Лучше выделить компонент  `App` в отдельный файл, чтобы не смешивать все в `index.js`

mikebars1995 commented on src/index.tsx:
```diff
@@ -13,19 +21,60 @@ const domNode = document.getElementById('root') as HTMLDivElement;
+
 	return (
 		<main
 			className={clsx(styles.main)}
```
**Можно лучше**

* `clsx` можно не использовать, если только **один** селектор передается

mikebars1995 commented on src/index.tsx:
```diff
@@ -13,19 +21,60 @@ const domNode = document.getElementById('root') as HTMLDivElement;
+		console.log(options);
+	}
+
+	function handleSetArticleState() {
```
**Нужно исправить**

* Вся логика формы должна быть описана внутри `ArticleParamsForm`, чтобы не засорять код `App` и пропсы `ArticleParamsForm`

mikebars1995 commented on src/components/article-params-form/ArticleParamsForm.tsx:
```diff
@@ -1,17 +1,128 @@
+							title='Применить'
+							htmlType='submit'
+							type='apply'
+							onClick={props.applySettings}
```
**Нужно исправить**

**Обработчик сабмита** нужно навешивать только на тег `form` с событием `submit`, а не на кнопку сабмита с событием `click`, так как сабмит формы происходит ещё при нажатии `Enter`, и он не будет работать, если навесить обработчик клика на кнопку только.

mikebars1995 commented on src/components/article-params-form/ArticleParamsForm.tsx:
```diff
@@ -1,17 +1,128 @@
+		}
+	}
+
+	useEffect(() => {
```
**Нужно исправить**

* Нужно **останавливать эффект**, если форма закрыта, чтобы не навешивать обработчик 

```js
if (!isOpen) return;
```

mikebars1995 commented on src/components/article-params-form/ArticleParamsForm.tsx:
```diff
@@ -1,17 +1,128 @@
+};
+
+export const ArticleParamsForm = (props: ArticleParamsProps) => {
+	const [isOpen, setIsOpen] = useState(false);
```
**Нужно исправить**

* `[isOpen, setIsOpen]` нужно назвать `[isMenuOpen, setIsMenuOpen]`, чтобы было понятно, что тут открываете именно **меню**

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:

* Хорошо реализовали форму из предоставленных компонентов
* Отлично, что нет типов `any`
* Отлично типизировали проект


но есть некоторые недочеты:

* Вся логика формы должна быть описана внутри `ArticleParamsForm`, чтобы не засорять код `App` и пропсы `ArticleParamsForm`
* **Обработчик сабмита** нужно навешивать только на тег `form` с событием `submit`, а не на кнопку сабмита с событием `click`, так как сабмит формы происходит ещё при нажатии `Enter`, и он не будет работать, если навесить обработчик клика на кнопку только.
* Нужно **останавливать эффект**, если форма закрыта, чтобы не навешивать обработчик 
* `[isOpen, setIsOpen]` нужно назвать `[isMenuOpen, setIsMenuOpen]`, чтобы было понятно, что тут открываете именно **меню**

**Можно лучше**

* Лучше выделить компонент  `App` в отдельный файл, чтобы не смешивать все в `index.js`
* `clsx` можно не использовать, если только **один** селектор передается

SmolinIvan committed:
fix: review fixes
1.The App component has been moved to a separate file
2.The form closing functionality has been improved
3.The form logic has been fully moved to its component

SmolinIvan commented:
Добрый вечер! Замечания поправлены

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
